dev-libs/msgpack_c_cpp/msgpack_c_cpp-2.0.0.recipe

```diff
@@ -0,0 +1,94 @@
+SUMMARY="A serialization library"
+DESCRIPTION="An efficient binary serialization format, which lets you exchange data \
+among multiple languages like JSON, except that it's faster and smaller. \
+Small integers are encoded into a single byte while typical short strings \
+require only one extra byte in addition to the strings themselves."
+HOMEPAGE="http://msgpack.org/"
+COPYRIGHT="2010-2016 MessagePack Developers"
+LICENSE="Boost v1.0"
+REVISION="1"
+SOURCE_URI="https://github.com/msgpack/msgpack-c/archive/cpp-$portVersion.tar.gz"
+SOURCE_DIR="msgpack-c-cpp-$portVersion"
+CHECKSUM_SHA256="eb20b4bf15f20bad149ec82fffac74f16de2a8a797e321a3f8189d63263a511e"
+
+ARCHITECTURES="x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	msgpack_c_cpp$secondaryArchSuffix = $portVersion
+	lib:libmsgpackc$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	msgpack_c_cpp${secondaryArchSuffix}_devel = $portVersion
+	devel:libmsgpackc$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	msgpack_c_cpp$secondaryArchSuffix == $portVersion
+	devel:libz${secondaryArchSuffix}
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libz${secondaryArchSuffix}
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:doxygen
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	pkgconfig$secondaryArchSuffix
+	"
+
+PATCH()
+{
+	sed -i 's/-pthread/-lroot/g' "example/cpp03/CMakeLists.txt"
+}
+
+BUILD()
+{
+	export LDFLAGS="-lroot -lnetwork -lbsd"
+	export CFLAGS="-D_BSD_SOURCE"
+
+	mkdir -p build && cd build
+	cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_SKIP_RPATH=YES \
+		..
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $includeDir
+	mkdir -p $libDir
+
+	cd build
+	make install
+
+	mv $libDir/../libmsgpackc.* $libDir/.
+	mv $libDir/../pkgconfig $libDir/.
+	mv $prefix/include/* $includeDir/
```

Comment:
No need to keep trailing «**`/.`**» or «**`/`**». You can replace these lines by:
```
	mv $libDir/../libmsgpackc.* $libDir
	mv $libDir/../pkgconfig $libDir
	mv $prefix/include/* $includeDir
```

That said, if you patch some files and/or pass extra parameters to cmake it might be possible to make «**`make install`**» install everything to the correct destinations.